jerry-debugger/jerry_client_websocket.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+
+# Copyright JS Foundation and other contributors, http://js.foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import struct
+from jerry_client_tcp import Socket
+
+MAX_BUFFER_SIZE = 128
+WEBSOCKET_BINARY_FRAME = 2
+WEBSOCKET_FIN_BIT = 0x80
+
+class WebSocket(object):
+    def __init__(self, address, protocol=Socket()):
```

Comment:
Using the `Socket()` as default argument is not correct. The `protocol` will have a reference to the created object and if the WebSocket is constructed multiple times the same socket will be used in both cases.
example.:
```
a = WebSocket()
b = WebSocket()
a.protocol == b.protocol
```

We should use `None` as default and add an check in the constructor to create the socket object.